atomic_reactor/build.py

```diff
@@ -54,37 +54,31 @@ def _ensure_not_built(self):
 
 
 class BuildResult(object):
-    def __init__(self, command_result, image_id=None):
-        """ when build fails, image_id is None """
-        self.command_result = command_result
+    def __init__(self, logs=None, fail_reason=None, image_id=None):
+        """
+        :param logs: iterable of log lines (without newlines)
+        :param fail_reason: str, description of failure or None if successful
+        :param image_id: str, ID of built container image
+        """
+        assert bool(fail_reason) != bool(image_id)  # must provide one, not both
```

Comment:
nitpick: add a more informative assert error, smth like 'Either of `fail_reason` or `image_id` should be provided, not both`